File: src/main/java/org/fb/manifest/KdiffApplication.java
package org.fb.manifest;

import javafx.application.Application;
import javafx.application.Platform;
import javafx.stage.Stage;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.context.ConfigurableApplicationContext;

public class KdiffApplication extends Application {

    private ConfigurableApplicationContext context;

    public void init() {
        context = new SpringApplicationBuilder()
                .sources(Main.class)
                .run();
    }

    public void start(Stage primaryStage) {
        context.publishEvent(new StageReadyEvent(primaryStage));
    }

    public void stop() {
        context.close();
        Platform.exit();
    }
}

File: src/main/kotlin/org/fb/kdiff/KdiffApplication.kt
package org.fb.kdiff

import javafx.application.Application
import javafx.application.Platform
import javafx.stage.Stage
import org.springframework.boot.builder.SpringApplicationBuilder
import org.springframework.context.ConfigurableApplicationContext

class KdiffApplication : Application() {

    private var context: ConfigurableApplicationContext? = null

    override fun init() {
        context = SpringApplicationBuilder()
                .sources(Main::class.java)
                .run()
    }

    override fun start(primaryStage: Stage) {
        context!!.publishEvent(StageReadyEvent(primaryStage))
    }

    override fun stop() {
        context!!.close()
        Platform.exit()
    }
}